dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -260,36 +253,55 @@ protected void generateNodeConfig(String logDir) {
         int port = PortHelper.getRandomPort();
         String endpoint = addr + ":" + port;
         String clientPort = " -client-port=";
-        String locators = "-locators=";
-        String locatorHost = null;
+        String locators = " -locators=";
+        String locPortString = " -peer-discovery-port=";
+        String locatorsList = null;
         String dirPath = snappyTest.getLogDir();
         String nodeLogDir = null;
         String timeStatistics = " -enable-time-statistics=" + SnappyPrms.getTimeStatistics() + " -statistic-archive-file=";
         String logLevel = " -log-level=" + SnappyPrms.getLogLevel();
+        String hostnameForPrimaryLocator = null;
         switch (snappyNode) {
             case LOCATOR:
-                nodeLogDir = HostHelper.getLocalHost() + " -dir=" + dirPath + clientPort + port + timeStatistics + "snappylocator.gfs" + logLevel;
-                SnappyBB.getBB().getSharedMap().put("locatorHost", HostHelper.getLocalHost());
-                SnappyBB.getBB().getSharedMap().put("locatorPort", Integer.toString(port));
+                int locPort;
+                do locPort = PortHelper.getRandomPort();
+                while (locPort < 0 && locPort > 65535);
```

Comment:
it can be simply replaced to while (false) or may you want to check something else